crates/zksync/compiler/src/zksolc/compile.rs

```diff
@@ -281,32 +284,40 @@ impl ZkSolc {
     /// versioned sources. These modified values can be accessed after the compilation process
     /// for further processing or analysis.
     pub fn compile(&mut self) -> Result<(ProjectCompileOutput, ContractBytecodes)> {
+        //TODO: understand how to set are_libaries_missing flag or if it should be set always
+        self.config.settings.are_libraries_missing = true;
         let mut displayed_warnings = HashSet::new();
         let mut data = BTreeMap::new();
         // Step 1: Collect Source Files
         let sources = self.get_versioned_sources().wrap_err("Cannot get source files")?;
         let mut contract_bytecodes = BTreeMap::new();
 
+        // Step 2: Prepare cache for contracts
+        let mut cached_contracts: HashMap<PathBuf, (Option<Vec<u8>>, String)> = HashMap::new();
+
+        // Step 3: Check missing libraries
+        // Map from (contract_path, contract_name) -> missing_libraries
+        let mut all_missing_libraries: HashMap<(String, String), HashSet<String>> = HashMap::new();
+
         // Step 2: Compile Contracts for Each Source
-        for (solc, version) in sources {
+        for version in sources.values() {
```

Comment:
Hmm I think we are introducing a regression here - we want to use this `solc` version...